particles/Services/particles/js/ImageClassifier.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+defineParticle(({DomParticle, log, html, resolver}) => {
+
+  const template_ = html`
+<div>
+  <h2>Classification with a generic image classifier</h2>
+  <img style="max-width: 240px;" src="{{imageUrl}}" alt="Image to classify"><br>
+  <div id="progress-bar">{{progress}}</div>
+  <div>
+    <div>Label: </span><span>{{label}}</div>
+    <div>Confidence: </span><span>{{probability}}</div>
+  </div>
+</div>
+  `;
+
+  const url = resolver(`ImageClassifier/../../assets/waltbird.jpg`);
+  const modelUrl = 'https://tfhub.dev/google/imagenet/mobilenet_v1_100_224/classification/1';
+  const labelUrl = resolver(`ImageClassifier/../../assets/ImageNetLabels.txt`);
+
+  return class extends DomParticle {
+    get template() {
+      return template_;
+    }
+
+    update({}, state) {
+      // TODO(sjmiles): update() is called during SpecEx, while
+      // render() is not. We'll put our processing code in render()
+      // to avoid being expensive at SpecEx time.
+    }
+
+    render({}, state) {
+      // formerly update
+      if (!state.loaded) {
+        state.loaded = true;
+        this.loadModel(modelUrl, {fromTFHub: true});
+      }
+      this.getModelLabels();
+
+      // render proper
+      let {response} = state;
+      response = response || {label: '<working>', probability: '<working>'};
+      return {
+        label: response.className,
+        probability: response.probability,
+        imageUrl: url
+      };
+    }
+
+    onLoadProgress(fraction) {
+      const prog = fraction < 1.0 ? fraction * 100.0 : fraction;
```

Comment:
It looks like this is allowing either passing say 50% or 0.5, do we actually need to support both?

Reply:
I wanted to force the fraction into a percentage. 

I'm going to get rid of load progress for now.